pkg/admission/validate_namespacestore.go

```diff
@@ -2,272 +2,107 @@ package admission
 	admissionv1 "k8s.io/api/admission/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// const configuration values for the validation checks
-const (
-	MaximumMountPathLength = 63
-)
-
-// NamespaceStoreValidator is the context of a namespacestore validation
-type NamespaceStoreValidator struct {
-	NamespaceStore *nbv1.NamespaceStore
-	Logger         *logrus.Entry
-	arRequest      *admissionv1.AdmissionReview
-	arResponse     *admissionv1.AdmissionReview
-}
-
 // NewNamespaceStoreValidator initializes a BackingStoreValidator to be used for loading and validating a namespacestore
-func NewNamespaceStoreValidator(arRequest admissionv1.AdmissionReview) *NamespaceStoreValidator {
-	nsv := &NamespaceStoreValidator{
-		arRequest: &arRequest,
+func NewNamespaceStoreValidator(arRequest admissionv1.AdmissionReview) *ResourceValidator {
+	nsv := &ResourceValidator{
+
 		Logger:    logrus.WithField("admission namespacestore validation", arRequest.Request.Namespace),
+		arRequest: &arRequest,
+		arResponse: &admissionv1.AdmissionReview{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "AdmissionReview",
+				APIVersion: "admission.k8s.io/v1",
+			},
+			Response: &admissionv1.AdmissionResponse{
+				UID:     arRequest.Request.UID,
+				Allowed: true,
+				Result: &metav1.Status{
+					Message: "allowed",
+				},
+			},
+		},
 	}
 	return nsv
 }
 
 // ValidateNamespaceStore call appropriate validations based on the operation
-func (nsv *NamespaceStoreValidator) ValidateNamespaceStore() admissionv1.AdmissionReview {
-	nsv.arResponse = &admissionv1.AdmissionReview{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "AdmissionReview",
-			APIVersion: "admission.k8s.io/v1",
-		},
-		Response: &admissionv1.AdmissionResponse{
-			UID:     nsv.arRequest.Request.UID,
-			Allowed: true,
-			Result: &metav1.Status{
-				Message: "allowed",
-			},
-		},
-	}
-
+func (nsv *ResourceValidator) ValidateNamespaceStore() admissionv1.AdmissionReview {
 	switch nsv.arRequest.Request.Operation {
 	case admissionv1.Create:
-		nsv.ValidateCreate()
+		nsv.ValidateCreateNS()
 	case admissionv1.Update:
-		nsv.ValidateUpdate()
+		nsv.ValidateUpdateNS()
 	case admissionv1.Delete:
-		nsv.ValidateDelete()
+		nsv.ValidateDeleteNS()
 	default:
 		nsv.Logger.Error("Failed to identify namespacestore operation type")
 	}
 	return *nsv.arResponse
 }
 
-// SetValidationResult responsible of assinging the return values of a validation into the response appropriate fields
-func (nsv *NamespaceStoreValidator) SetValidationResult(isAllowed bool, message string) {
-	nsv.arResponse.Response.Allowed = isAllowed
-	nsv.arResponse.Response.Result.Message = message
-}
-
 // DeserializeNS extract the namespacestore object from the request
-func (nsv *NamespaceStoreValidator) DeserializeNS(rawNS []byte) *nbv1.NamespaceStore {
+func (nsv *ResourceValidator) DeserializeNS(rawNS []byte) *nbv1.NamespaceStore {
 	NS := nbv1.NamespaceStore{}
 	if err := json.Unmarshal(rawNS, &NS); err != nil {
```

Comment:
I know it's not related to your changes but notice that the parameter rawNS can be extracted from nsv, so it's redundant

Reply:
but sometime I use this function to deserialize `arRequest.Request.Object.Raw` and sometimes I use it to deserialize `bsv.arRequest.Request.OldObject.Raw`.
I keep it `ResourceValidator` struct function to print the appropriate error if the `json.Unmarshal()` failed.